opencolorio_config_aces/config/generation/factories.py

```diff
@@ -0,0 +1,570 @@
+                produce_transform(from_reference),
+                ocio.VIEWTRANSFORM_DIR_FROM_REFERENCE)
+
+    view_transform.setName(name)
+
+    if family is not None:
+        view_transform.setFamily(family)
+
+    for category in categories:
+        view_transform.addCategory(category)
+
+    if description is not None:
+        view_transform.setDescription(description)
+
+    return view_transform
+
+
+@required('OpenColorIO')
+def look_factory(name,
+                 process_space=None,
+                 description=None,
+                 forward_transform=None,
+                 inverse_transform=None,
+                 base_look=None,
+                 **kwargs):
+    """
+    *OpenColorIO* look factory.
+
+    Parameters
+    ----------
+    name : unicode
+        *OpenColorIO* look name.
+    process_space : unicode, optional
+        *OpenColorIO* look process space, e.g. *OpenColorIO* colorspace or role
+        name.
+    description : unicode, optional
+        *OpenColorIO* look description.
+    forward_transform : dict or object, optional
+        *To Reference* *OpenColorIO* look transform.
+    inverse_transform : dict or object, optional
+        *From Reference* *OpenColorIO* look transform.
+    base_look : dict or ViewTransform, optional
+        *OpenColorIO* base look inherited for initial attribute values.
+
+    Other Parameters
+    ----------------
+    \\**kwargs : dict, optional
+        Keywords arguments.
+
+    Returns
+    -------
+    Look
+        *OpenColorIO* look.
+    """
+
+    import PyOpenColorIO as ocio
+
+    if process_space is None:
+        process_space = ocio.ROLE_SCENE_LINEAR
+
+    if base_look is not None:
+        if isinstance(base_look, Mapping):
+            base_look = look_factory(**base_look)
+
+        look = base_look
+    else:
+        look = ocio.Look()
+
+        look.setProcessSpace(process_space)
+
+        if forward_transform is not None:
+            look.setTransform(produce_transform(forward_transform))
+
+        if inverse_transform is not None:
+            look.setInverseTransform(produce_transform(inverse_transform))
+
+    look.setName(name)
+
+    if description is not None:
+        look.setDescription(description)
+
+    return look
+
+
+@required('OpenColorIO')
+def transform_factory_default(**kwargs):
+    """
+    *OpenColorIO* default transform factory that produces an *OpenColorIO*
+    transform according to given ``name`` keyword argument.
+
+    Other Parameters
+    ----------------
+    name : unicode
+        *OpenColorIO* transform class/type name, e.g. ``CDLTransform``.
+    \\**kwargs : dict, optional
+        Setter keywords arguments. They are converted to *camelCase* with *set*
+        prepended, e.g. `base_colorspace` is transformed into
+        `setBaseColorspace`.
+
+    Returns
+    -------
+    object
+        *OpenColorIO* transform.
+    """
+
+    import PyOpenColorIO as ocio
+
+    transform = getattr(ocio, kwargs.pop('name'))()
```

Comment:
This is gonna fail for "LogCameraTransform"; possibly others. The constructor requires a `linSideBreak` value.

On the other hand,  the "LogAffineTransform" doesn't accept an initial value for `base` during instantiation -- you have to use the `setBase`  instance method afterwards.

Damned if you do, damned if you don't.

Reply:
That's fine, we can handle those special cases one-by-one.